lib/charting.rb

```diff
@@ -0,0 +1,6 @@
+# If code uses the old constant name:
+#   * Rails will autoload it and start here.
+#   * We assign the old toplevel constant to the new constant.
+#   * We can't include rails deprecate_constant globally, so we use ruby's.
+Charting = ManageIQ::Reporting::Charting
+Object.deprecate_constant :Charting
```

Comment:
Reviewers, let me explain this.  Rails has `deprecate_constant `that you can use by including `ActiveSupport::Deprecation::DeprecatedConstantAccessor` in your class/module.  This is great and works well in the old `ReportFormatter`, see the lib/manageiq/reporting/formatter.rb.  The major benefit is it tells you what is deprecated and what's the new behavior to use.

Unfortunately, you can't include that module globally, and we have two global constants, `Charting` and `ReportFormatter` to deprecate.  Ruby has `deprecate_constant`, which we can call on Object to ensure any references to `Charting` `ReportFormatter` at the top level do raise deprecations.  The ruby version doesn't need to include any modules but it also just tells you it's deprecated without any suggestions as to the solution.  I think this is the best we can do.  Maybe someone knows something better?

So, here, we setup the old Constant, set to the new one and deprecate the old one.